_finalproj/src/stage1.lua

```diff
@@ -40,37 +40,458 @@ function Stage1:init(def)
+        end)
+    elseif self.mech4counter == 2 then
+        self.mech4counter = self.mech4counter + 1
+        Timer.after(t*2, function()
+            local aoe1= AOE{
+                shape = 'triangle',
+            
+                --coordinates 
+                x = VIRTUAL_WIDTH/2,
+                y = VIRTUAL_HEIGHT/2,
+            
+                x2 = 0, --triangle only 
+                y2 = 0, --triangle only   
+                x3 = 0, --triangle only
+                y3 = VIRTUAL_HEIGHT, --triangle only
+    
+                damage = 3, --how much damage this AOE will inflict
+                snaptime = 1, -- time telegraph will show/when the snapshot occurs
+                player = self.player
+    
+            }
+            self.stage:addAOE(aoe1)
+        end)
+    elseif self.mech4counter == 3 then
+        self.mech4counter = self.mech4counter + 1
+        Timer.after(t*3, function()
+            local aoe1= AOE{
+                shape = 'triangle',
+            
+                --coordinates 
+                x = VIRTUAL_WIDTH/2,
+                y = VIRTUAL_HEIGHT/2,
+            
+                x2 = 0, --triangle only 
+                y2 = 0, --triangle only   
+                x3 = VIRTUAL_WIDTH, --triangle only
+                y3 = 0, --triangle only    
+    
+                damage = 3, --how much damage this AOE will inflict
+                snaptime = 1, -- time telegraph will show/when the snapshot occurs
+                player = self.player
+    
+            }
+            self.stage:addAOE(aoe1)
+        end)
+    elseif self.mech4counter == 4 then
+        self.mech4counter = self.mech4counter + 1
+        Timer.after(t*4, function()
+            local aoe1= AOE{
+                shape = 'triangle',
+            
+                --coordinates 
+                x = VIRTUAL_WIDTH/2,
+                y = VIRTUAL_HEIGHT/2,
+            
+                x2 = VIRTUAL_WIDTH, --triangle only 
+                y2 = 0, --triangle only   
+                x3 = VIRTUAL_WIDTH, --triangle only
+                y3 = VIRTUAL_HEIGHT, --triangle only
+    
+                damage = 3, --how much damage this AOE will inflict
+                snaptime = 1, -- time telegraph will show/when the snapshot occurs
+                player = self.player
+    
+            }
+            self.stage:addAOE(aoe1)
+
+            Timer.after(t, function()
+                self.mech4counter = 1
+            end)
+    
+        end)
+    end
+
+
+end
+
+
+
+function Stage1:update(dt)
+    self.stage:update(dt)
+    --TODO: phase changes 
+    self.boss.phase = 2
+    if self.boss.phase == 1 then --TENNIS BALLS BARRAGE
+        self:mech1()
+    elseif self.boss.phase == 2 then --TRAFFIC
+        self:mech2()
+    elseif self.boss.phase == 3 then --BOMB TOSS
+        self:mech3()
+    elseif self.boss.phase == 4 then --DUSTSTORM
+        self:mech4()
+    elseif self.boss.phase == 5 then --DUSTSTORM
```

Comment:
Is this a typo? Are 4 and 5 both meant to be duststoms?

Reply:
yeah that was copy pasted. ill correct it later